src/views/UserEdit/UserEditHelpers.js

```diff
@@ -28,18 +28,35 @@ export function resourcesLoaded(obj, exceptions = []) {
  * @param {object} res HTTP response object from a rejected Promise
  * @param {function} sendCallout function reference, likely this.context.sendCallout
  */
-export function showErrorCallout(res, sendCallout) {
-  res.text()
-    .then(body => {
-      sendCallout({
-        type: 'error',
-        timeout: 0,
-        message: (
-          <div>
-            <div><strong><FormattedMessage id="ui-users.errors.permissionChangeFailed" /></strong></div>
-            <div>{body}</div>
-          </div>
-        )
-      });
+export async function showErrorCallout(res, sendCallout) {
+  const calloutOptions = { type: 'error', timeout: 0 };
+  try {
+    const contentType = res.headers.get('content-type');
+    let errorMessage = '';
+
+    if (contentType && contentType.includes('application/json')) {
+      const json = await res.json();
+      errorMessage = json?.errors?.map(e => e.message)?.join(', ');
+    } else {
+      errorMessage = await res.text();
+    }
+
+    const isUserExists = errorMessage?.includes('username already exists');
```

Comment:
As I understand it, this is an internal error that occurs on the BE side. This error is returned to the FE but it can be changed by BE team without any notification.
On my opinion we must avoid being tied to messages from BE. Instead, the BE should extend the returned object (with errors) and pass a key referring to which we can accurately verify the type of error.

Reply:
BE already works in this way, I would agree that we need a json response, but mod-users is not our module to change approach of response type and it returns string error for permissions error - I have no idea what behind it

Reply:
@alisher-epam probably it makes sense to create a ticket (mod-users) for Volaris team to make some consistency - please do it

cc @Dmitriy-Litvinenko

Reply:
@NikitaSedyx @Dmitriy-Litvinenko I have created a ticket for the Volaris team. Feel free to add more info. https://issues.folio.org/browse/MODUSERS-382